.github/workflows/cross-branch-migrations.yml

```diff
@@ -0,0 +1,144 @@
+name: Cross-branch migration tests
+
+on:
+  pull_request:
+    types: [opened, synchronize, reopened, ready_for_review]
+    paths:
+      - "resources/migrations/**"
+
+concurrency:
+  group: ${{ github.workflow }}-${{ github.head_ref && github.ref || github.run_id }}
+  cancel-in-progress: true
+
+jobs:
+  migrations-h2-ee:
+    runs-on: ubuntu-22.04
+    timeout-minutes: 3
+    env:
+      CI: 'true'
+      MB_DB_TYPE: h2
+    steps:
+      - uses: actions/checkout@v4
+      - uses: ./.github/actions/cross-branch-migration-test
+
+  migrations-postgres-12-ee:
+    runs-on: ubuntu-22.04
+    timeout-minutes: 3
```

Comment:
Apparently I can't set `timeout-minutes` on the health check steps in the action due to a GitHub limitation so I've set it here instead. It would be nicer if we could kill it automatically when the health check passes but I'm not sure how to do that and don't see any existing patterns like that